File: backend/app/utils/municipios.py
import unicodedata
from typing import List, Optional

MUNICIPIOS_CERCANOS: List[str] = ["bello", "medellin", "itagui", "envigado"]
DEPARTAMENTO_VALIDO: str = "antioquia"

def normalizar_texto(texto: str) -> str:
    """
    Convierte texto a minúsculas, remueve espacios en los extremos y elimina acentos/tildes.
    Ejemplo: 'Itagüí' -> 'itagui', 'Medellín' -> 'medellin', 'Antioquia' -> 'antioquia'
    """
    if not texto:
        return ""
    texto_limpio = texto.strip().lower()
    nfkd_form = unicodedata.normalize('NFD', texto_limpio)
    return "".join([c for c in nfkd_form if not unicodedata.combining(c)])

def es_municipio_cercano(municipio: str, departamento: Optional[str] = None) -> bool:
    """
    Determina si un municipio dado pertenece a los municipios cercanos y
    que el departamento sea estrictamente Antioquia.
    """
    if departamento is not None and normalizar_texto(departamento) != DEPARTAMENTO_VALIDO:
        return False
    municipio_normalizado = normalizar_texto(municipio)
    return municipio_normalizado in MUNICIPIOS_CERCANOS

File: frontend/src/utils/municipios.js
export const MUNICIPIOS_CERCANOS = ['bello', 'medellin', 'itagui', 'envigado'];
export const DEPARTAMENTO_VALIDO = 'antioquia';

/**
 * Normaliza un texto convirtiéndolo a minúsculas, quitando espacios extremos y removiendo acentos/tildes.
 * @param {string} texto 
 * @returns {string}
 */
export function normalizarMunicipio(texto) {
  if (!texto) return '';
  return texto
    .trim()
    .toLowerCase()
    .normalize('NFD')
    .replace(/[\u0300-\u036f]/g, '');
}

/**
 * Evalúa si el municipio ingresado corresponde estrictamente a uno de los municipios cercanos
 * (Bello, Medellín, Itagüí o Envigado) Y si el departamento ingresado es Antioquia.
 * @param {string} municipio 
 * @param {string} [departamento]
 * @returns {boolean}
 */
export function esMunicipioCercano(municipio, departamento) {
  if (departamento !== undefined && normalizarMunicipio(departamento) !== DEPARTAMENTO_VALIDO) {
    return false;
  }
  const municipioLimpio = normalizarMunicipio(municipio);
  return MUNICIPIOS_CERCANOS.includes(municipioLimpio);
}
